alchemist/alchemist-maps/src/main/java/it/unibo/alchemist/model/implementations/movestrategies/speed/TraceDependantSpeed.java

```diff
@@ -37,7 +38,7 @@ public TraceDependantSpeed(final MapEnvironment<T> e, final Node<T> n, final Rea
         env = e;
         node = n;
         reaction = r;
```

Comment:
The bare objects are already checked for null values by calling LangUtils.requireNonNull(e, n, r). Do you prefer to change it and call requireNonNull the same way it is now done in the new line? I don't think it would make that much difference.